src/Expr.cc

```diff
@@ -2601,14 +2601,9 @@ VectorValPtr index_slice(VectorVal* vect, int _first, int _last) {
 
     zeek_int_t first = get_slice_index(_first, len);
     zeek_int_t last = get_slice_index(_last, len);
-    zeek_int_t sub_length = last - first;
 
-    if ( sub_length >= 0 ) {
-        result->Resize(sub_length);
```

Comment:
I'm going to drop this commit. The pre `Resize()` does allocate the required memory at once while the `Assign()` may reallocate a few times. There's also `Reserve()` which would allow replacing `Assign()` with `Append()`, but more generally wonder if this could work and be even faster when switched to ZVal's.

See also. https://github.com/zeek/zeek/pull/3478#discussion_r1415692021

Reply:
Thanks, makes sense.